src/spawn_sys.rs

```diff
@@ -0,0 +1,450 @@
+                        commands.spawn((
+                            SpriteBundle {
+                                transform: Transform::from_xyz(x, y, 0.0)
+                                    .with_scale(GLOBAL_RESCALE_V * 4.0)
+                                    .with_rotation(Quat::from_rotation_z(-3.0 * PI / 4.0)),
+                                texture: asset_server.load(ship_sprite_path),
+                                ..default()
+                            },
+                            Velocity {
+                                velocity: Vec3 {
+                                    x: 0.0,
+                                    y: 0.0,
+                                    z: 0.0,
+                                },
+                            },
+                            load_boss_ship(),
+                        ));
+                    }
+                    _ => {
+                        println!("ship_type not recognized!")
+                    }
+                }
+            }
+        }
+    }
+}
+
+pub fn spawn_asteroid_system(
+    mut commands: Commands,
+    window_query: Query<&Window, With<PrimaryWindow>>,
+    asset_server: Res<AssetServer>,
+) {
+    let window = window_query.get_single().unwrap();
+    let mut rng = thread_rng();
+    for _ in 0..10 {
+        let random_x = rng.gen::<f32>() * window.width();
+        let random_y = rng.gen::<f32>() * window.height();
+        let asteroid_rescaler = rng.gen::<f32>();
+
+        commands.spawn((
+            SpriteBundle {
+                transform: Transform::from_xyz(random_x, random_y, 0.0)
+                    .with_scale(GLOBAL_RESCALE_V * asteroid_rescaler),
+                texture: asset_server.load("sprites/environmental/meteorGrey_big1.png"),
+                ..default()
+            },
+            Asteroid {},
+            Clipping {
+                cd_timer: Timer::from_seconds(0.15, TimerMode::Once),
+            },
+            CollisionBox {
+                shape: Shape::Circle,
+                width_radius: 42.0 * GLOBAL_RESCALE_C * asteroid_rescaler,
+                height: 42.0 * GLOBAL_RESCALE_C * asteroid_rescaler,
+            },
+            Health { value: 10.0 },
+            Mass {
+                value: 10000.0 * asteroid_rescaler,
+            },
+            Velocity {
+                velocity: Vec3 {
+                    x: rng.gen::<f32>() * 0.1,
+                    y: rng.gen::<f32>() * 0.1,
+                    z: 0.0,
+                },
+            },
+            EntityType::Asteroid,
+        ));
+    }
+}
+
+pub fn despawn_dead_system(
+    mut commands: Commands,
+    entity_query: Query<(Entity, &Health, &EntityType)>,
+    asset_server: Res<AssetServer>,
+) {
+    for (entity, health, et) in entity_query.iter() {
+        // If an entity's health has dropped to or below 0, despawn it.
+        if health.value <= 0.0 {
+            if *et == EntityType::Ship || *et == EntityType::Missile {
```

Comment:
Why not use your marker components?

To avoid having to check a long list of markers, you could create additional markers for common properties. For example, create a marker for anything that should be able to die and add it to all the relevant entities. You can then just have a query filter for that.